### PR TITLE
Advanced item level versioning for DS 7

### PR DESCRIPTION
**Pull request by parphis**
## References
* Fixes https://jira.lyrasis.org/browse/DS-4586
* Fixes #7919 

## Description
This PR is the modification of the #3376 one in order to apply it for the version 7.x

As per the linked ticket
> We were thinking of improving item versioning. Our aim was to:
>
> 1. provide a persistent Handle for each item version
> 2. provide a persistent Handle that references always the most recent version

> While the current DSpace Handle versioning scheme allows (1), it does not provide (2). Basically our idea is the scheme of how  arXiv.org provides persistent item references.

## Instructions for Reviewers
With these modifications we may add a configuration property to the basic config: `initialVersionSuffix`.
> If this option is missing or if it has an empty value then the item versioning acts like before. However, if it has a string of ".1" then the Handle of the very first version of the item will get a version suffix.

Most of the modifications have been applied on the `HandleServiceImpl.java` in order to add a new parameter for the function `findHandleInternal`. With the new boolean argument we can make difference between the calls:
 1. Are we trying to resolve a handle - use true.
 2. Are we checking the existence of a handle during id creation - use false.

We also has changed the `createId` function to use the `initialVersionSuffix` and to determine wheter the new handle id will be used for an item or any other resource?

A new function `findByVersion` within `HandleDAO` class aims the modified functionality in the `findHandleInternal`. It returns a handle list within the Version table based on the given handle.

Applying our modifications result the followings when using the item versioning feature (assuming the `initialVersionSuffix` is set to _.1_):
 1. Every new item gets a versioned handle like 123456789/1.1
 2. Every new version added will get new version numbers increased by one.
 3. Linking to a handle with a specific version number renderers the exact item. However linking to an item without any version suffix (like 123456789/1) returns the most recent version of the item.

In case of missing option `initialVersionSuffix` or leaving it empty means the original item versionig will be used.

## Checklist

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

**Comment by tdonohue** [comment]
@abollini : as your team is working on Item Level versioning support for 7.1 (in #3371), I'd appreciate your feedback on this suggestion/improvement on how version numbers are assigned to items.  So, I'm adding you as a reviewer (feel free to delegate to another on your team if needed).  This may not necessarily need to be in 7.1, but I wanted to alert you to this work in case you found it to be a worthwhile "quick fix".

**Review by abollini — CHANGES_REQUESTED** [review]
Sorry to come back to this PR so late. I found an incomplete review that I have never published.
At the current stage this PR requires some update to be considered
- we need additional javadoc on the service methods to understand the differences between the `findByVersion` and `findByPrefix` methods as they seem to be duplicate
- the `INITIAL_VERSION_SUFFIX` cannot be really anything else than .1 or the empty string so it should be better eventually to have a boolean attribute that switch from the current behavior to the new one
- test must be provided to proof that the new configuration works as expected

**Comment by tdonohue** [comment]
@parphis : FYI, It looks like your most recent commit (to add integration tests) is failing tests

**Comment by parphis** [comment]
@tdonohue Thanks, I'm working on it continuosly locally.

**Comment by parphis** [comment]
@abollini I changed the code as per you requests/suggestions:

 - I create integration tests to see whether the handle creation and resolving processes are working well.
 - I removed the constant and changed the code based on this modification.
 - I also removed the `findByVersion` function and changed the `findByHandle` one in order to better express changes.

Please let me know if you find other problems. I have a bit of experience in Java and integration testing methods but I am trying to do the best.

@tdonohue You wrote in an e-mail in the last weeks that we should notice you until 1th of August if we need to add new features for the version 7.4. So I would like to ask you to add this PR to the version 7.4. I try to do my best to correct any errors before the release date of October. Thank you.

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+
+        item = ItemBuilder.createItem(context, collection)
+                          .withTitle("Public item with first .1 suffix")
+                          .withIssueDate("2022-07-15")
+                          .withAuthor("Smith, Donald").withAuthor("Doe, John")
+                          .withSubject("ExtraEntry")
+                          .build();
+
+        System.out.println("Item handle is: ");
+        System.out.println(item.getHandle());
```
Comment (abollini) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java:183]: please remove

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+
+        getClient(token).perform(get("/api/versioning/versions/" + version2nd.getID()))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$", Matchers.is(VersionMatcher.matchEntry(version2nd))))
+            .andExpect(jsonPath("$._links.versionhistory.href", Matchers.allOf(Matchers.containsString(
+                                "api/versioning/versions/" + version2nd.getID() + "/versionhistory"))))
+            .andExpect(jsonPath("$._links.item.href", Matchers.allOf(Matchers.containsString(
+                                "api/versioning/versions/" + version2nd.getID() + "/item"))))
+            .andExpect(jsonPath("$._links.self.href", Matchers.allOf(Matchers.containsString(
+                                "api/versioning/versions/" + version2nd.getID()))));
```
Comment (abollini) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java:220]: you need to check that the new versions will get .2 and .3

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+                                              context, item.getHandle());
+        // get URL without ".1" suffix
+        String resolvedHandleWithoutSuffix = HandleServiceFactory.getInstance().getHandleService().resolveToURL(
+                                                context, item.getHandle().split("\\.")[0]);
+        String resolvedVersionHandle = HandleServiceFactory.getInstance().getHandleService().resolveToURL(
+                                          context, version1st.getItem().getHandle());
+
+        System.out.println("resolved handle with suffix: " + resolvedHandleWithSuffix);
+        System.out.println("resolved handle without suffix: " + resolvedHandleWithoutSuffix);
+
```
Comment (abollini) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java:255]: please remove

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+        // get URL with ".1" suffix
+        resolvedHandleWithSuffix = HandleServiceFactory.getInstance().getHandleService().resolveToURL(
+                                              context, item.getHandle());
+        // get URL without ".1" suffix
+        resolvedHandleWithoutSuffix = HandleServiceFactory.getInstance().getHandleService().resolveToURL(
+                                                context, item.getHandle().split("\\.")[0]);
+
+        System.out.println("resolved handle with suffix: " + resolvedHandleWithSuffix);
+        System.out.println("resolved handle without suffix: " + resolvedHandleWithoutSuffix);
+
```
Comment (abollini) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java:272]: please remove

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java [hunk 5]
```diff
@@ -0,0 +1,384 @@
...
+                                "api/versioning/versions/" + version2nd.getID() + "/versionhistory"))))
+            .andExpect(jsonPath("$._links.item.href", Matchers.allOf(Matchers.containsString(
+                                "api/versioning/versions/" + version2nd.getID() + "/item"))))
+            .andExpect(jsonPath("$._links.self.href", Matchers.allOf(Matchers.containsString(
+                                "api/versioning/versions/" + version2nd.getID()))));
+    }
+
+    @Test
+    public void resolveToURLWithFirstSuffixTest() throws Exception {
+        System.out.println("resolveToURLTest");
```
Comment (abollini) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java:225]: please remove

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java [hunk 6]
```diff
@@ -0,0 +1,384 @@
...
+          .andExpect(jsonPath("$._links.self.href", Matchers.allOf(Matchers.containsString(
+                              "api/versioning/versions/" + version2nd.getID()))));
+    }
+
+    /*
+     * Test the item and its versions built with initialVersionSuffix = true
+     */
+    @Test
+    public void itemWithFirstVersionSuffixTest() throws Exception {
+        System.out.println("itemWithVersionSuffixTest");
```
Comment (abollini) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java:169]: please remove

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java [hunk 7]
```diff
@@ -0,0 +1,384 @@
...
+
+        baseUrl = configurationService.getProperty("dspace.ui.url") + "/handle/";
+    }
+
+    /*
+     * Test the item and its versions built with initialVersionSuffix = false
+     */
+    @Test
+    public void itemWithoutVersionSuffixTest() throws Exception {
+        System.out.println("itemWithoutVersionSuffixTest");
```
Comment (abollini) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java:110]: please remove

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java [hunk 8]
```diff
@@ -0,0 +1,384 @@
...
+                          .withIssueDate("2022-07-15")
+                          .withAuthor("Smith, Donald").withAuthor("Doe, John")
+                          .withSubject("ExtraEntry")
+                          .build();
+
+        version1st = VersionBuilder.createVersion(
+            context, item, "First version of the item without .1 suffix")
+                .build();
+
+        System.out.println("hello from resolveToURLTest: " + item.getHandle());
```
Comment (abollini) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java:242]: please remove

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java [hunk 9]
```diff
@@ -0,0 +1,384 @@
...
+        System.out.println("resolved handle with suffix: " + resolvedHandleWithSuffix);
+        System.out.println("resolved handle without suffix: " + resolvedHandleWithoutSuffix);
+
+        assertEquals(resolvedHandleWithSuffix, baseUrl + item.getHandle());
+        assertEquals(resolvedHandleWithoutSuffix, resolvedHandleWithoutSuffix);
+    }
+
+    @Test
+    public void resolveToURLWithoutFirstSuffixTest() throws Exception {
+        System.out.println("resolveToURLTest");
```
Comment (abollini) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java:279]: please remove

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java [hunk 10]
```diff
@@ -0,0 +1,384 @@
...
+        context.turnOffAuthorisationSystem();
+
+        item = ItemBuilder.createItem(context, collection)
+                          .withTitle("Public item with first .1 suffix")
+                          .withIssueDate("2022-07-15")
+                          .withAuthor("Smith, Donald").withAuthor("Doe, John")
+                          .withSubject("ExtraEntry")
+                          .build();
+
+        System.out.println("hello from resolveToURLTest: " + item.getHandle());
```
Comment (abollini) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java:292]: please remove

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java [hunk 11]
```diff
@@ -0,0 +1,384 @@
...
+
+        // get URL with ".1" suffix
+        String resolvedHandleWithSuffix = HandleServiceFactory.getInstance().getHandleService().resolveToURL(
+                                              context, item.getHandle());
+        // get URL without ".1" suffix
+        String resolvedHandleWithoutSuffix = HandleServiceFactory.getInstance().getHandleService().resolveToURL(
+                                                context, item.getHandle().split("\\.")[0]);
+
+        System.out.println("resolved handle with suffix: " + resolvedHandleWithSuffix);
+        System.out.println("resolved handle without suffix: " + resolvedHandleWithoutSuffix);
```
Comment (abollini) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java:302]: please remove

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java [hunk 12]
```diff
@@ -0,0 +1,384 @@
...
+        configurationService.setProperty("dspace.initialVersionSuffix", true);
+        // get URL with ".1" suffix
+        resolvedHandleWithSuffix = HandleServiceFactory.getInstance().getHandleService().resolveToURL(
+                                              context, item.getHandle());
+        // get URL without ".1" suffix
+        resolvedHandleWithoutSuffix = HandleServiceFactory.getInstance().getHandleService().resolveToURL(
+                                                context, item.getHandle().split("\\.")[0]);
+
+        System.out.println("resolved handle with suffix: " + resolvedHandleWithSuffix);
+        System.out.println("resolved handle without suffix: " + resolvedHandleWithoutSuffix);
```
Comment (abollini) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java:317]: please remove

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java [hunk 13]
```diff
@@ -0,0 +1,384 @@
...
+        context.turnOffAuthorisationSystem();
+
+        item = ItemBuilder.createItem(context, collection)
+                          .withTitle("Public item with first .1 suffix")
+                          .withIssueDate("2022-07-15")
+                          .withAuthor("Smith, Donald").withAuthor("Doe, John")
+                          .withSubject("ExtraEntry")
+                          .build();
+
+        System.out.println("hello from resolveUrlToHandleTest: " + item.getHandle());
```
Comment (abollini) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java:338]: please remove

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java [hunk 14]
```diff
@@ -0,0 +1,384 @@
...
+        System.out.println("resolved handle with suffix: " + resolvedHandleWithSuffix);
+        System.out.println("resolved handle without suffix: " + resolvedHandleWithoutSuffix);
+
+        assertEquals(resolvedHandleWithSuffix, baseUrl + item.getHandle());
+        assertEquals(resolvedHandleWithoutSuffix, baseUrl + item.getHandle().split("\\.")[0]);
+    }
+
+    @Test
+    public void resolveUrlToHandleTest() throws Exception, SQLException {
+        System.out.println("resolveUrlToHandleTest");
```
Comment (abollini) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java:325]: please remove

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java [hunk 15]
```diff
@@ -0,0 +1,384 @@
...
+
+        // get URL with ".1" suffix
+        String resolvedHandleWithSuffix = HandleServiceFactory.getInstance().getHandleService().resolveUrlToHandle(
+                                              context, baseUrl + item.getHandle());
+        // get URL without ".1" suffix
+        String resolvedHandleWithoutSuffix = HandleServiceFactory.getInstance().getHandleService().resolveUrlToHandle(
+                                                context, baseUrl + item.getHandle().split("\\.")[0]);
+
+        System.out.println("resolved handle with suffix: " + resolvedHandleWithSuffix);
+        System.out.println("resolved handle without suffix: " + resolvedHandleWithoutSuffix);
```
Comment (abollini) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java:348]: please remove

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java [hunk 16]
```diff
@@ -0,0 +1,384 @@
...
+
+        assertEquals(resolvedHandleWithSuffix, item.getHandle());
+        assertEquals(resolvedHandleWithoutSuffix, item.getHandle().split("\\.")[0]);
+    }
+
+    @Test
+    public void resolveToObjectTest() throws Exception, SQLException {
+        System.out.println("resolveToObjectTest");
```
Comment (abollini) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java:356]: please remove

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java [hunk 17]
```diff
@@ -0,0 +1,384 @@
...
+        context.turnOffAuthorisationSystem();
+
+        item = ItemBuilder.createItem(context, collection)
+                          .withTitle("Public item with first .1 suffix")
+                          .withIssueDate("2022-07-15")
+                          .withAuthor("Smith, Donald").withAuthor("Doe, John")
+                          .withSubject("ExtraEntry")
+                          .build();
+
+        System.out.println("hello from resolveToObjectTest: " + item.getHandle());
```
Comment (abollini) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java:369]: please remove

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java [hunk 18]
```diff
@@ -0,0 +1,384 @@
...
+
+        System.out.println("resolved handle with suffix: " + resolvedHandleWithSuffix);
+        System.out.println("resolved handle without suffix: " + resolvedHandleWithoutSuffix);
+
+        assertEquals(resolvedHandleWithSuffix, baseUrl + item.getHandle());
+        assertEquals(resolvedHandleWithoutSuffix, resolvedHandleWithoutSuffix);
+    }
+
+    @Test
+    public void resolveToURLWithoutFirstSuffixTest() throws Exception {
```
Comment (abollini) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java:278]: you should perform a test also with two versions to verify that the .1 version resolve correctly and the most recent handle resolve correctly too.

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java [hunk 19]
```diff
@@ -0,0 +1,384 @@
...
+
+        System.out.println("resolved handle with suffix: " + resolvedHandleWithSuffix);
+        System.out.println("resolved handle without suffix: " + resolvedHandleWithoutSuffix);
+
+        assertEquals(resolvedHandleWithSuffix, baseUrl + item.getHandle());
+        assertEquals(resolvedHandleWithoutSuffix, baseUrl + item.getHandle().split("\\.")[0]);
+    }
+
+    @Test
+    public void resolveUrlToHandleTest() throws Exception, SQLException {
```
Comment (abollini) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java:324]: you should perform a test also with two versions to verify that the .1 version resolve correctly and the most recent handle resolve correctly too.

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java [hunk 20]
```diff
@@ -0,0 +1,384 @@
...
+
+        // get dso with ".1" suffix
+        DSpaceObject resolvedDsoWithSuffix = HandleServiceFactory.getInstance().getHandleService().resolveToObject(
+                                              context, item.getHandle(), true);
+        // get dso without ".1" suffix
+        DSpaceObject resolvedDsoWithoutSuffix = HandleServiceFactory.getInstance().getHandleService().resolveToObject(
+                                                context, item.getHandle().split("\\.")[0], true);
+
+        System.out.println("resolved dso with suffix: " + resolvedDsoWithSuffix.getHandle());
+        System.out.println("resolved dso without suffix: " + resolvedDsoWithoutSuffix.getHandle());
```
Comment (abollini) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java:379]: please remove

**Review by abollini — CHANGES_REQUESTED** [review]
Hi @parphis thanks for your work here. I have made some comment inline in the code but just before to submit my review I have noted that we already have in the codebase an alternative implementation of the VersionedHandleProvider that looks to implement exactly the same scenario that you want to achieve. Moreover, this implementation follow the main suggestion that I have put in the review of your code, the "most recent version handle" is stored in the database.
Can you take a look to `org.dspace.identifier.VersionedHandleIdentifierProviderWithCanonicalHandles` to see if I'm right? if this is the case we should keep the IT that you have introduced in this PR and verify them against this implementation

**Comment by tdonohue** [comment]
@parphis : It looks like this PR is stalled.  Are you still planning to work on this for our upcoming 7.4 release (due in early Oct)? If so, we'd need updates to this in the next few weeks to help move it forward.  If not, we can also reschedule for 7.5 (due in Feb)

**Comment by parphis** [comment]
@tdonohue as per the notes from @abollini I have tested the versioning with the recommended settings (with `VersionedHandleIdentifierProviderWithCanonicalHandles` uncommented) last week and it has been determined that it fulfills our requirements for the versioning of the handles. This means that changes I have made on the code can be dropped.

Currently, I am fine tuning the created integration test in order to apply the suggestions of @abollini 

Please let me know if I should open a new PR for this new integration test or leave it here? Thank you!

File: dspace-api/src/main/java/org/dspace/handle/dao/impl/HandleDAOImpl.java [hunk 21]
```diff
@@ -63,20 +63,46 @@ public List<Handle> getHandlesByDSpaceObject(Context context, DSpaceObject dso)
...
+                "ORDER BY v.versionDate desc "
+                );
+
+            query.setParameter("handle", handle + "%");
+        } else {
+            query = createQuery(context,
+                "SELECT h " +
+                "FROM Handle h " +
+                "LEFT JOIN FETCH h.dso " +
+                "WHERE h.handle = :handle ");
```
[the remark this answers is not in the document]
Comment (parphis) [reply dspace-api/src/main/java/org/dspace/handle/dao/impl/HandleDAOImpl.java:92]: @abollini As I described below the `VersionedHandleIdentifierProviderWithCanonicalHandles` option does what we need regarding the versioning. So I close this PR and will open a new one with the modified integration tests. Thank you for your support.

**Comment by parphis** [comment]
>  if this is the case we should keep the IT that you have introduced in this PR and verify them against this implementation

@abollini @tdonohue 
I would like to run an integration test to see if the handle versioning is working well using the `org.dspace.identifier.VersionedHandleIdentifierProviderWithCanonicalHandles`.

I copy the `identifier-service.xml` into the folders `dspace-api/src/test/data/dspaceFolder/config/spring/api` and `dspace-server-webapp/src/test/data/dspaceFolder/config/spring/api` with the uncommented lines
```XML
<bean id="org.dspace.identifier.HandleIdentifierProvider" class="org.dspace.identifier.VersionedHandleIdentifierProviderWithCanonicalHandles" scope="singleton">
        <property name="configurationService" ref="org.dspace.services.ConfigurationService"/>
    </bean>
```
I checked the file `dspace-server-webapp/target/testing/dspace/config/spring/api/identifier-service.xml` and found that the uncommented lines are there.
But when I run the integration test and print out the generated handles I see the default versioning functionality. That is, the origin item gets a handle without a version suffix, the second version gets a .2, the third gets a .3 etc. So it seems that the test does not use the `VersionedHandleIdentifierProviderWithCanonicalHandles` class as expected.
Could you please support me on how should I set up the IT in order to use the canonical handles version? Is there any function in the test environment to modify this configuration at load time (like the configuration service)? (Sorry I am not using the Java/Spring tests regularly so I have limited knowledge on it). Thanks!

**Pull request closed without merging**